src/codegen.py

```diff
@@ -97,6 +99,14 @@ def generate_files_for_template(env, template_file, input_files, output_dir):
             except OSError:
                 print("Could not make output directory", output_dir)
                 sys.exit(1)
+        elif _CLEAN:
+            # Removes any files in the directory
```

Comment:
I think the implementation could be a bit simpler. You wouldn't need to run the code once per file. You could run `shutil.rmtree` on the entire output directory ahead of creating the output directory.

Reply:
.